src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -1956,6 +1968,568 @@ public Task OpenAsync(CancellationToken token)
+                    return Task.CompletedTask;
+                }
+
+                public Task CloseOutputSessionAsync(CancellationToken token)
+                {
+                    return CloseOutputSessionAsync();
+                }
+            }
+
+            public class ServerSecurityDuplexSessionChannelDispatcher : ServerSecurityDuplexSessionChannel, IDuplexSessionChannel, IServiceChannelDispatcher
+            {
+                private readonly IServiceProvider _serviceProvider;
+                private volatile IServiceChannelDispatcher _channelDispatcher;
+
+                public ServerSecurityDuplexSessionChannelDispatcher(
+                    SecuritySessionServerSettings settings,
+                    SecurityContextSecurityToken sessionToken,
+                    object listenerSecurityState, SecurityListenerSettingsLifetimeManager settingsLifetimeManager
+                    , IChannel channel, EndpointAddress address)
+                    : base(settings,
+                          sessionToken, listenerSecurityState, settingsLifetimeManager, address)
+                {
+                    IncomingChannel = (IDuplexSessionChannel)channel;
+                    _serviceProvider = IncomingChannel.GetProperty<IServiceScopeFactory>().CreateScope().ServiceProvider;
+                }
+
+                public IDuplexSessionChannel IncomingChannel { get; set; }
+                public IServiceChannelDispatcher ChannelDispatcher { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+                public event EventHandler Closed;
+                public event EventHandler Closing;
+                public event EventHandler Faulted;
+                public event EventHandler Opened;
+                public event EventHandler Opening;
+
+                public Task DispatchAsync(RequestContext context)
+                {
+                    return DispatchAsync(context.RequestMessage);
+                }
+
+                public Task DispatchAsync(Message message)
+                {
+                    DuplexSessionRequestContext duplexSessionRequestContext = new
+                        DuplexSessionRequestContext(IncomingChannel, message);
+                    RequestContext context =  ReceiveRequest(duplexSessionRequestContext);
+                    return _channelDispatcher.DispatchAsync(context == null ? null :context.RequestMessage);
+                }
+
+                public T GetProperty<T>() where T : class
+                {
+                    T tObj = _serviceProvider.GetService<T>();
+                    if (tObj == null)
+                    {
+                        return IncomingChannel.GetProperty<T>();
+                    }
+                    else
+                    {
+                        return tObj;
+                    }
+                }
+
+                public Task OpenAsync()
+                {
```

Comment:
Why is this OpenAsync a no-op but the one with a TimeSpan specified has an implementation? This OpenAsync and the one 2 methods below which calls this are different than the overridden one.